src/pages/iou/ReceiptSelector/index.js

```diff
@@ -0,0 +1,200 @@
+import Receipt from '../../../libs/actions/Receipt';
+import useWindowDimensions from '../../../hooks/useWindowDimensions';
+import useLocalize from '../../../hooks/useLocalize';
+
+const propTypes = {
+    /** Information shown to the user when a receipt is not valid */
+    receiptModal: PropTypes.shape({
+        isAttachmentInvalid: PropTypes.bool,
+        attachmentInvalidReasonTitle: PropTypes.string,
+        attachmentInvalidReason: PropTypes.string,
+    }),
+
+    /** Route params */
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            iouType: PropTypes.string,
+            reportID: PropTypes.string,
+        }),
+    }),
+
+    /** The report on which the request is initiated on */
+    report: reportPropTypes,
+
+    /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
+    iou: PropTypes.shape({
+        id: PropTypes.string,
+        amount: PropTypes.number,
+        currency: PropTypes.string,
+        participants: PropTypes.arrayOf(
+            PropTypes.shape({
+                accountID: PropTypes.number,
+                login: PropTypes.string,
+                isPolicyExpenseChat: PropTypes.bool,
+                isOwnPolicyExpenseChat: PropTypes.bool,
+                selected: PropTypes.bool,
+            }),
+        ),
+    }),
+
+    /** Current user personal details */
+    currentUserPersonalDetails: personalDetailsPropType,
+
+    /** Used by drag and drop to determine if we have a file dragged over the view */
+    isDraggingOver: PropTypes.bool,
+};
+
+const defaultProps = {
+    receiptModal: {
+        isAttachmentInvalid: false,
+        attachmentInvalidReasonTitle: '',
+        attachmentInvalidReason: '',
+    },
+    route: {
+        params: {
+            iouType: '',
+            reportID: '',
+        },
+    },
+    report: {},
+    iou: {
+        id: '',
+        amount: 0,
+        currency: CONST.CURRENCY.USD,
+        participants: [],
+    },
+    ...withCurrentUserPersonalDetailsDefaultProps,
+    isDraggingOver: false,
+};
+
+function ReceiptSelector({route, currentUserPersonalDetails, isDraggingOver, iou, report, receiptModal}) {
+    const iouType = useRef(lodashGet(route, 'params.iouType', ''));
+    const reportID = useRef(lodashGet(route, 'params.reportID', ''));
+    const isAttachmentInvalid = lodashGet(receiptModal, 'isAttachmentInvalid', false);
+    const attachmentInvalidReasonTitle = lodashGet(receiptModal, 'attachmentInvalidReasonTitle', '');
+    const attachmentInvalidReason = lodashGet(receiptModal, 'attachmentInvalidReason', '');
+    const [receiptImageTopPosition, setReceiptImageTopPosition] = useState(0);
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const {translate} = useLocalize();
+
+    /**
+     * Close the confirm modal.
+     */
+    const closeConfirmModal = useCallback(() => {
+        Receipt.clearUploadReceiptError();
+    }, []);
```

Comment:
Move outside

Reply:
I get this lint error: `ESLint: React Hook "useCallback" cannot be called at the top level. React Hooks must be called in a React function component or a custom React Hook function.(react-hooks/rules-of-hooks)`

Does this not need the useCallback hook?

Reply:
@AndrewGable yeah you don't need the `useCallback` hook here – instead of `closeConfirmModal` you can just pass `Receipt.clearUploadReceiptError` directly to `ConfirmModal` props